sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/client/XdsClient.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.implement.service.xds.client;
+
+import io.envoyproxy.envoy.service.discovery.v3.AggregatedDiscoveryServiceGrpc;
+import io.envoyproxy.envoy.service.discovery.v3.AggregatedDiscoveryServiceGrpc.AggregatedDiscoveryServiceStub;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryRequest;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryResponse;
+import io.grpc.ManagedChannel;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.stub.MetadataUtils;
+import io.grpc.stub.StreamObserver;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.sermant.core.common.LoggerFactory;
+import io.sermant.core.config.ConfigManager;
+import io.sermant.core.service.xds.config.XdsConfig;
+import io.sermant.core.utils.FileUtils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.net.ssl.SSLException;
+
+/**
+ * XdsClient
+ *
+ * @author daizhenyu
+ * @since 2024-05-09
+ **/
+public class XdsClient implements Closeable {
+    private static final Logger LOGGER = LoggerFactory.getLogger();
+
+    private XdsConfig config = ConfigManager.getConfig(XdsConfig.class);
+
+    private ManagedChannel channel;
+
+    /**
+     * construction method
+     */
+    public XdsClient() {
+        createChannel();
+    }
+
+    private void createChannel() {
+        if (config.isSecurityEnable()) {
+            try {
+                SslContext context = GrpcSslContexts.forClient()
+                        .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                        .build();
+                channel = NettyChannelBuilder.forTarget(config.getControlPlaneAddress())
+                        .sslContext(context)
+                        .build();
+            } catch (SSLException e) {
+                LOGGER.log(Level.SEVERE, "SSLException occurred when creating gRPC channel", e);
+            }
+        } else {
+            channel = NettyChannelBuilder.forTarget(config.getControlPlaneAddress())
+                    .usePlaintext()
+                    .build();
```

Comment:
you can change  chaining method calls of channel to multiple lines, just make before  and after `if ` the action is the same

Reply:
done